server/src/main/java/org/elasticsearch/script/field/vectors/BinaryDenseVector.java

```diff
@@ -37,11 +38,27 @@ public float[] getVector() {
         return decodedDocVector;
     }
 
+    @Override
+    public byte[] asBytes() {
+        throw new UnsupportedOperationException();
+    }
+
     @Override
     public float getMagnitude() {
         return VectorEncoderDecoder.getMagnitude(indexVersion, docVector);
     }
 
+    @Override
+    public double dotProduct(byte[] queryVector) {
+        ByteBuffer byteBuffer = wrap(docVector);
+
+        double dotProduct = 0;
+        for (float v : queryVector) {
+            dotProduct += byteBuffer.getFloat() * v;
+        }
```

Comment:
I think we should probably have a high-bandwidth discussion on how some of the byte/float calculations should interact. 

In this case `BinaryDenseVector` is the `float` version of this class. I'd like to rename it in a follow up to `FloatBinaryDenseVector` which would match `ByteBinaryDenseVector`, but didn't want to add to the already really high number of lines in this PR.

I guess the question I have is should this just be an `UnsupportedOperationException`? With scripting it's a bit strange because an unusual case could have a query across multiple indices where a field is a `float` vector and the same field in a different index is a `byte` vector.`

Reply:
Updated do stop mixed operations between `byte[]` and `float[]` vectors.